extensions/integration-tests/src/notebook.test.ts

```diff
@@ -14,136 +14,183 @@ import { sqlNotebookContent, writeNotebookToFile, sqlKernelMetadata, getFileName
+	invocationCount: number = 0;
+
+	private static IterationCount = 20;
+	private static ParallelCount = 1;
+
+	@stressify({ dop: NotebookTester.ParallelCount, iterations: NotebookTester.IterationCount })
+	async pySpark3NbTest(title: string): Promise<void> {
+		let notebook = await this.openNotebook(pySparkNotebookContent, pySpark3KernelMetadata, title + this.invocationCount++);
+		let cellOutputs = notebook.document.cells[0].contents.outputs;
+		let sparkResult = (<azdata.nb.IStreamResult>cellOutputs[3]).text;
+		assert(sparkResult === '2', `Expected spark result: 2, Actual: ${sparkResult}`);
 	}
 
-	return notebook;
-}
-async function verifyClearAllOutputs(notebook: azdata.nb.NotebookEditor) {
-	let cellWithOutputs = notebook.document.cells.find(cell => cell.contents && cell.contents.outputs && cell.contents.outputs.length > 0);
-	assert(cellWithOutputs !== undefined, 'Could not find notebook cells with outputs');
-	console.log('Before clearing cell outputs');
-	let clearedOutputs = await notebook.clearAllOutputs();
-	let cells = notebook.document.cells;
-	cells.forEach(cell => {
-		assert(cell.contents && cell.contents.outputs && cell.contents.outputs.length === 0, `Expected Output: 0, Actual: '${cell.contents.outputs.length}'`);
-	});
-	assert(clearedOutputs, 'Outputs of all the code cells from Python notebook should be cleared');
-	console.log('After clearing cell outputs');
+	@stressify({ dop: NotebookTester.ParallelCount, iterations: NotebookTester.IterationCount })
+	async python3ClearAllOutputs(title: string): Promise<void> {
+		let notebook = await this.openNotebook(pySparkNotebookContent, pythonKernelMetadata, title + this.invocationCount++);
+		await this.verifyClearAllOutputs(notebook);
+	}
+
+	@stressify({ dop: NotebookTester.ParallelCount, iterations: NotebookTester.IterationCount })
+	async python3NbTest(title: string): Promise<void> {
+		let notebook = await this.openNotebook(pySparkNotebookContent, pythonKernelMetadata, title + this.invocationCount++);
+		let cellOutputs = notebook.document.cells[0].contents.outputs;
+		console.log('Got cell outputs ---');
+		if (cellOutputs) {
+			cellOutputs.forEach(o => console.log(JSON.stringify(o, undefined, '\t')));
+		}
+		let result = (<azdata.nb.IExecuteResult>cellOutputs[0]).data['text/plain'];
+		assert(result === '2', `Expected python result: 2, Actual: ${result}`);
+	}
+
+	@stressify({ dop: NotebookTester.ParallelCount, iterations: NotebookTester.IterationCount })
+	async sqlNbClearAllOutputs(title: string): Promise<void> {
+		let notebook = await this.openNotebook(sqlNotebookContent, sqlKernelMetadata, title + this.invocationCount++);
+		await this.verifyClearAllOutputs(notebook);
+	}
+
+	@stressify({ dop: NotebookTester.ParallelCount, iterations: NotebookTester.IterationCount })
+	async sqlNbMultipleCellsTest(title: string): Promise<void> {
+		let notebook = await this.openNotebook(sqlNotebookMultipleCellsContent, sqlKernelMetadata, title + this.invocationCount++, true);
```

Comment:
Lots of these could be const instead. In general you should use const instead of let whenever possible (i.e. you don't need to modify the var after it's initialized)

Reply:
yes , I know but this was pre-existing code that I did not look into the details of to modify. I just used refactoring tools to wrap code blocks into methods.

Reply:
Agree with you. But it is pre-existing code, should get cleaned up independently.

---
In reply to: [287894538](https://github.com/microsoft/azuredatastudio/pull/5499#discussion_r287894538) [](ancestors = 287894538)